File: backend/src/main/kotlin/com/zdsj/affiliate/ActivityTags.kt
package com.zdsj.affiliate

/**
 * 活动标签清洗：过滤数据源占位标签，仅保留对用户有意义的营销/活动信息。
 */
object ActivityTags {

    /** 联盟通道降级或映射兜底时写入的内部标记，不应展示给用户 */
    private val INTERNAL = setOf(
        "京东联盟",
        "维易",
        "维易·京东联盟",
        "多多进宝",
    )

    fun sanitize(tags: List<String>): List<String> =
        tags.map { it.trim() }.filter { it.isNotBlank() && it !in INTERNAL }
}

File: miniprogram/utils/activityTags.js
/** 数据源占位标签，不应展示给用户 */
const INTERNAL_ACTIVITY_TAGS = new Set([
  '京东联盟',
  '维易',
  '维易·京东联盟',
  '多多进宝',
]);

/** 过滤内部占位标签，保留真实活动/营销标签 */
function sanitizeActivityTags(tags) {
  return (tags || [])
    .map((t) => String(t).trim())
    .filter((t) => t && !INTERNAL_ACTIVITY_TAGS.has(t));
}

module.exports = { sanitizeActivityTags, INTERNAL_ACTIVITY_TAGS };
